msg.go

```diff
@@ -403,35 +403,31 @@ Loop:
 	return string(s), off1, nil
 }
 
-func packTxt(txt []string, msg []byte, offset int, tmp []byte) (int, error) {
+func packTxt(txt []string, msg []byte, offset int) (int, error) {
 	var err error
+	if offset >= len(msg) {
+		return offset, ErrBuf
+	}
 	if len(txt) == 0 {
-		if offset >= len(msg) {
-			return offset, ErrBuf
-		}
 		msg[offset] = 0
 		return offset, nil
```

Comment:
Is this the correct behavior when len(txt) == 0?

Is this any different from
```
if len(txt) == 0 {
    return offset, nil
}
```
without doing the 0 assignment to the returned offset?